File: project/SeongBeen-onboarding/src/main/java/com/group/surveyapp/domain/entity/Question.java
package com.group.surveyapp.domain.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 설문조사와 다대일 관계
    @ManyToOne
    @JoinColumn(name = "survey_id")
    private Survey survey;

    private String name; // 항목 이름

    private String description; // 항목 설명

    @Enumerated(EnumType.STRING)
    private QuestionType type; // SHORT, LONG, SINGLE, MULTI

    private boolean required; // 필수 여부

    // 선택형인 경우 후보값 (ElementCollection)
    @ElementCollection
    @CollectionTable(
            name = "question_candidates",
            joinColumns = @JoinColumn(name = "question_id")
    )
    @Column(name = "candidate")
    private List<String> candidates;

}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/api/survey/domain/Question.kt
package com.wlghsp.querybox.api.survey.domain

import jakarta.persistence.*
import support.domain.BaseRootEntity

@Entity
class Question(
    var name: String,
    var description: String,

    @Enumerated(EnumType.STRING)
    var type: QuestionType,

    var required: Boolean,

    @ElementCollection
    @CollectionTable(name = "question_option", joinColumns = [JoinColumn(name = "question_id")])
    val options: List<Option>? = null,

    id: Long = 0L,
) : BaseRootEntity<Question>(id) {

    init {
        require(name.isNotBlank()) { "문항 이름은 비어 있을 수 없습니다." }

        if (type == QuestionType.SINGLE_CHOICE || type == QuestionType.MULTIPLE_CHOICE) {
            require(!options.isNullOrEmpty()) { "선택형 문항에는 옵션이 반드시 있어야 합니다." }

            val distinctCheckTarget = options!!
            require(distinctCheckTarget.distinctBy { it.value }.size == distinctCheckTarget.size) {
                "옵션 값은 중복될 수 없습니다."
            }
        } else {
            require(options.isNullOrEmpty()) { "주관식 문항은 옵션을 가질 수 없습니다." }
        }
    }
}
